cmd/soroban-cli/src/config/upgrade_check.rs

```diff
@@ -0,0 +1,83 @@
+use crate::config::locator;
+use jsonrpsee_core::Serialize;
+use semver::Version;
+use serde::Deserialize;
+use serde_json;
+use std::fs;
+
+const FILE_NAME: &str = "upgrade_check.json";
+
+/// The `UpgradeCheck` struct represents the state of the upgrade check.
+/// This state is global and stored in the `upgrade_check.json` file in
+/// the global configuration directory.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct UpgradeCheck {
+    /// The timestamp of the latest check for a new version of the CLI.
+    pub latest_check_time: u64,
```

Comment:
One last idea. Use `chrono::DateTime<chrono::offset::Utc>` directly. This way again don't have to deal with converting to a primitive and can compare directly with `chrono::Utc::now()`

`now - MINIMUM_CHECK_INTERVAL >= stats.latest_check_time`. 

Small nit but makes it a little cleaner.

Reply:
Thanks for spotting this, changed in c967a6102d07ed6160dd3bbfbdffa2ab160c9ba4.